packages/flame/lib/src/experimental/geometry/shapes/rounded_rectangle.dart

```diff
@@ -171,6 +171,24 @@ class RoundedRectangle extends Shape {
     return result;
   }
 
+  @override
+  Vector2 nearestPoint(Vector2 point) {
+    final xx = (point.x).clamp(_left, _right);
+    final yy = (point.y).clamp(_top, _bottom);
+    final result = Vector2(xx, yy);
+    if (containsPoint(result)) {
+      return result;
+    }
+    final center = result
+      ..setValues(
+        result.x <= _left + _radius ? _left + _radius : _right - _radius,
+        result.y <= _top + _radius ? _top + _radius : _bottom - _radius,
+      );
+    final vectorToPoint = point - center;
+    final distance = vectorToPoint.length;
```

Comment:
It should be possible to use `distanceTo` here right?

Reply:
Ah right, `vectorToPoint` is used later

Reply:
(also refactored to use static temporary vectors)